hashing/fnv/fnv.go

```diff
@@ -27,11 +27,17 @@ func (f Fnv) Compute(s string) []byte {
 // EmptyHash returns the fnv128a hash of the empty string
 func (f Fnv) EmptyHash() []byte {
 	if len(fnvEmptyHash) == 0 {
-		fnvEmptyHash = f.Compute("")
+		f.setEmptyHash()
 	}
 	return fnvEmptyHash
 }
 
+func (f Fnv) setEmptyHash() {
+	h := fnv.New128a()
+	_, _ = h.Write([]byte(""))
+	fnvEmptyHash = h.Sum(nil)
```

Comment:
We need to address the slight possibility of concurrent write here. As discussed, we might have constructors for these and make all function receivers as pointers.

Reply:
removed. will do in another task